File: cpp/42b_1872A_TwoVessels.cpp
#include<bits/stdc++.h>

int main(){
    int t, a, b, c;
    float mid;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>a>>b>>c;
        mid = (a + b) / 2.0;
        //std::cout<<mid<<" "<<ceil(abs(a - mid) / c)<<" "<<ceil(abs(b - mid) / c)<<"\n"; 
        ans.push_back(std::min(ceil(abs(a - mid) / c), ceil(abs(b - mid) / c)));
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/42b_1872A_TwoVessels.py
from math import ceil as ceil

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        a, b, c = input().split()
        a, b, c = int(a), int(b), int(c)
        mid = (a + b) / 2
        ans.append(min(ceil(abs(a - mid) / c), ceil(abs(b - mid) / c)))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()